analyzers/src/SonarAnalyzer.Common/AnalysisContext/SonarAnalysisContext.cs

```diff
@@ -108,19 +109,19 @@ public void RegisterCodeBlockStartAction<TSyntaxKind>(GeneratedCodeRecognizer ge
         analysisContext.RegisterSymbolAction(
             c => Execute<SonarSymbolReportingContext, SymbolAnalysisContext>(new(this, c), action, null), symbolKinds);
 
-    public void RegisterNodeAction<TSyntaxKind>(GeneratedCodeRecognizer generatedCodeRecognizer, Action<SonarSyntaxNodeReportingContext> action, params TSyntaxKind[] syntaxKinds)
-        where TSyntaxKind : struct =>
-        analysisContext.RegisterSyntaxNodeAction(
-            c => Execute<SonarSyntaxNodeReportingContext, SyntaxNodeAnalysisContext>(new(this, c), action, c.Node.SyntaxTree, generatedCodeRecognizer), syntaxKinds);
+    public void RegisterTreeAction(GeneratedCodeRecognizer generatedCodeRecognizer, Action<SonarSyntaxTreeReportingContext> action) =>
```

Comment:
RegisterTreeAction didn't change, but it appears in the diff because of the change in the methods' order. Could you consider restoring the old order so the diff history is more straightforward?